Bottom: b5d09fb9afc86c0bd242fca9a00b1ad8db186d76
Top:    7abf17eb4d91e19582471e43dd856ccec8678c5b
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 13:20:49 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/mount_autofs.c b/modules/mount_autofs.c
index 8c1e600..00a91e2 100644
--- a/modules/mount_autofs.c
+++ b/modules/mount_autofs.c
@@ -178,12 +178,14 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name,
 
 	argc = 1;
 
-	if (!(info = parse_map_type_info(what))) {
-		error(ap->logopt, MODPREFIX "failed to parse map info");
-		master_free_mapent(entry);
-		return 1;
+	if (what) {
+		if (!(info = parse_map_type_info(what))) {
+			error(ap->logopt, MODPREFIX "failed to parse map info");
+			master_free_mapent(entry);
+			return 1;
+		}
+		argv[0] = info->map;
 	}
-	argv[0] = info->map;
 
 	if (options) {
 		p = options;
@@ -192,7 +194,11 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name,
 				*p = '\0';
 				p++;
 			}
-			argv[argc++] = p;
+			if (strcmp(fstype, "autofs") &&
+			    strncmp(p, "hosts", 5))
+				info = parse_map_type_info("hosts:");
+			else
+				argv[argc++] = p;
 		} while ((p = strchr(p, ',')) != NULL);
 	}
 	argv[argc] = NULL;
